### ISSUE TITLE
exporting optimized extrinsics

### ISSUE THREAD

**Issue by nitinagarwal** [NVlabs/instant-ngp#1128]
Hi @Tom94,

I wanted to export the optimized extrinsics for a dataset. However, when I use the `get_camera_extrinsics` from the python API, I believe its simply exporting the original camera extrinsics. Do you have any thoughts or suggestions on this?

**Comment by Tom94** [comment]
It works for me. Did you forget to enable extrinsics training (it's off by default)? Or did you load a snapshot (optimized extrinsics are not saved into snapshots)?

**Comment by nitinagarwal** [comment]
Thanks @Tom94. I was loading from the snapshot, didn't realize you weren't saving those as well. 
Another question I had was, have you experimented with how much error in the extrinsics can be optimized by instantNGP. (I have already read your response from #69 and was curious if you had any update)

**Comment by Tom94** [comment]
Yeah, storing extrinsics/intrinsics in the snapshot is unfortunately a bit non-trivial, because snapshots are meant to be independent of the dataset. For example, you can load one dataset and a snapshot that was trained on another. And then finetune the NeRF on the new dataset. Overwriting the extrinsics of the new dataset with those from the old one would be wrong -- and detecting such cases automatically would involve extra machinery. I'm not opposed to such a feature and would accept PRs.

RE what kind of error can be optimized by i-ngp: this is ultimately governed by local minima in gradient descent, which are plentiful in image-space. Still, there a surprising degree of error correction that can be accomplished. See the following video, where I perturbed _all_ the training poses of the lego dataset by ~10%, yet it still recovers completely.

https://user-images.githubusercontent.com/4923655/208869299-016a6cb7-8e3c-4119-a2bb-eee9b1bf1e5c.mp4

For larger errors in the poses, you could try seeding the optimization with lots of initial guesses over the sphere and to re-sample the best ones to get around local minima, like we do here: https://pnerfp.github.io

**Comment by nitinagarwal** [comment]
Thanks @Tom94 for the detailed reply and a Happy New Year to you!
